src/services/anchor-service.ts

```diff
@@ -80,25 +80,27 @@ export class AnchorService {
   /**
    * Creates anchors for client requests
    */
-  public async anchorRequests(): Promise<void> {
+  public async anchorRequests(triggeredByAnchorEvent = false): Promise<void> {
     // TODO: Remove this after restart loop removed as part of switching to go-ipfs
     // Skip sleep for unit tests
     if (process.env.NODE_ENV != 'test') {
       logger.imp('sleeping one minute for ipfs to stabilize')
       await Utils.delay(1000 * 60)
     }
 
-    logger.imp('Anchoring pending requests...')
-    // We try to fill our batch with 2^merkleDepthLimit streams at the leaf nodes of the merkle tree.
-    // To make sure we find enough requests on unique streamids to fill the batch, we pull in 100x
-    // the number of requests as the number of unique streams we are looking for. If we *still*
-    // don't find enough unique streams in all those requests, then we wind up with an under-full
-    // batch.  More likely, we pull in more requests than we need, but the remainder will just
-    // be picked up by the next anchor batch.
-    const streamLimit = Math.pow(2, this.config.merkleDepthLimit)
-    const requestLimit = streamLimit * 100
+    logger.imp('Anchoring ready requests...')
     logger.debug(`Loading requests from the database`)
-    const requests: Request[] = await this.requestRepository.findNextToProcess(requestLimit)
+
+    const readyRequests = await this.requestRepository.findByStatus(RS.READY)
+
+    if (!triggeredByAnchorEvent && readyRequests.length === 0) {
```

Comment:
This `anchorRequests` function now seems to be doing two fairly different things - anchoring ready requests, but also setting pending requests to ready (when running in the current mode where anchors aren't scheduled separately).  I feel like we should be able to break this up better and make more clear separation of responsibilities and control flow.  I'd love to see this function renamed to `anchorReadyRequests` and always be able to assume that the requests it needs to anchor are already in the READY status.  To do that, we'd need to move this block here up higher in the call stack.  I think higher up you have more natural codepath divergence between whether we're running in "v1" or "v2" mode, where this logic may fit more naturally.

Feel free to disagree though, at this point you're more familiar with the structure of this codebase than I am

Reply:
I'm imagining code that looks something like:

```
if (!triggeredByEvent) {
  markRequestsReadyIfNeeded()
}
anchorReadyRequests()
```

Reply:
maybe actually this function could *be* the higher-level thing that runs the pseudocode I just outlined, but with `_anchorRequests` renamed to `_anchorReadyRequests`, and a helper function added for cleaning up the `if (!triggeredByEvent)` code for making requests ready

Reply:
yup i like this. I'm going to leave `_anchorRequests` as is because its a helper function and is used heavily in the tests. But i've added an `anchorReadyRequests` and `anchorRequests`.  `anchorRequests` retains its original functionality where requests start as PENDING. `anchorReadyRequests` assume the PENDING requests have already moved to READY. When we are cleaning this up later, we just need to remove `anchorRequests` and replace it with `anchorReadyRequests. 

When we trigger an anchor via the server, is that anchoring READY requests or PENDING requests. We can leave that decision to later when we fully transition to CAS V2.